### DISCUSSION TITLE
Why is LayoutTransformControl's DecimalsAfterRound hard coded to 4?

### DISCUSSION THREAD

**Discussion by BobLd** [AvaloniaUI/Avalonia#16499] [Q&A] [answered]
Hi all,

I asked this question on the Telegram channel but I think it got lost.

Does anybody know if there is any particular reason why `LayoutTransformControl.cs`' `DecimalsAfterRound` value is set to 4?

When zooming into (very) large objects, the rounding creates issues when using the Matrix (I created my own version with the value set to 8).

Do you think this could be made either a parameter, or increased?

**Comment by maxkatz6** [comment]
It's just always been this way since https://github.com/AvaloniaUI/Avalonia/pull/468/files, not really documented why.

**Reply by maxkatz6** [reply]
Actually, this const is inherited from Silverlight, see https://github.com/microsoftarchive/SilverlightToolkit/blob/master/Release/Silverlight3/Source/Controls.Layout.Toolkit/LayoutTransformer/LayoutTransformer.cs#L84

**Reply by maxkatz6** [reply]
No idea if we still need it though.
cc @grokys

**Reply by BobLd** [reply]
@maxkatz6 thanks a lot for the feedback. I see the following comment in the Silverlight code:
```csharp
  // Note: AcceptableDelta and DecimalsAfterRound work around double arithmetic rounding issues on Silverlight.
```

Ideally, the whole rounding logic could be removed - not sure double arithmetic issue is only in Silverlight though. From my tests it seems safe to increase from 4 to 8

**Reply by maxkatz6** [reply]
It might be related to the fact that Silverlight had to run on various platforms

**Reply by maxkatz6** [reply, marked as answer]
We can increase it to 8, keeping it in 11.2 builds without backporting. And see if it regresses anything.
Can you create a pull request? Ideally with a comment explaining where this number comes from, not like before :)

**Reply by BobLd** [reply]
@maxkatz6 PR created here https://github.com/AvaloniaUI/Avalonia/pull/16583
